src/main/daml/Daml/Finance/Holding/Util.daml

```diff
@@ -90,6 +92,7 @@ releaseImpl :
   , HasToInterface t Base.I
   ) => Optional Base.Lock -> (Optional Base.Lock -> t) -> Base.Release -> Update (ContractId Base.I)
 releaseImpl currentLock setLock Base.Release{context} = do
+  assertMsg "Must be locked" $ isSome currentLock
```

Comment:
What was the behaviour before the check?

Reply:
I added this as a sanity check for the `releaseImpl` (in case it would be used elsewhere). 

The behaviour from before does not actually change, but I think it adds clarity: 
The controllers for the `Release` are the `lockers`, and in the case `lockers` are empty the choice can't be exercised (it is not possible to exercise a choice having an empty controller set). The added check makes it more explicit I would argue.